### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by nweiyue**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1074,22 +1141,21 @@ Use case ends.
...
-1. Should not contain very large file sizes (JAR files - 100Mb, PDF files - 15Mb/file).
+1. This product should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+1. This product should be able to hold up to 500 students and 500 sessions without a noticeable sluggishness in performance for typical usage.
+1. This product should be optimised for users who can type regular English text (i.e. not code, not system admin commands) fast and prefer typing over other means of input (e.g., mouse).
+1. This product should be for a single user and should not be a multi-user product.
+1. This product should not depend on remote servers.
+1. This product should be able to function without being connected to a network.
+1. This product should be able to work without requiring an installer.
+1. Data generated by the product should be stored locally in human-editable file.
+1. This product Should not contain very large file sizes (JAR files - 100Mb, PDF files - 15Mb/file).
```
Comment (erisjacey) [inline docs/DeveloperGuide.md]: Maybe "Should" should be de-capitalized here?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1133,26 +1199,49 @@ testers are expected to do more *exploratory* testing.
...
+  1. Test case: `addstu n/Cathy Duigan m/A1123456X e/cathyduigan@u.nus.edu t/helpful`<br>
+       Expected: Similar to previous.
+       
+  1. Test case: `addstu m/A2123456X n/Elbert Foo e/elbertfoo@u.nus.edu`<br>
+       Expected: Similar to previous.
+       
+  1. Test case: `addstu n/Gina Ho m/A3123456X`<br>
+     Expected: No student is added. Error message indicating an invalid command format is shown in the result box.
+     
+  1. Test case: `addstu n/Gina Ho e/ginaho@u.nus.edu`<br>
```
Comment (erisjacey) [inline docs/DeveloperGuide.md]: Maybe pictures could be provided for at least one positive test case and one negative test case each?

**Review by erisjacey — APPROVED** [review]
LGTM, just a few nitpicks

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (nweiyue) [reply docs/DeveloperGuide.md]: Good spot! Thank you.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (nweiyue) [reply docs/DeveloperGuide.md]: Nice suggestion! Will do!

**Pull request merged**